test/serverExplorer.test.ts

```diff
@@ -0,0 +1,140 @@
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import { ServersViewTreeDataProvider } from '../src/serverExplorer';
+import { RSPClient, Protocol} from 'rsp-client';
+import { EventEmitter, window } from 'vscode';
+
+const expect = chai.expect;
+chai.use(sinonChai);
+
+suite('Server explorer', () => {
+    let sandbox: sinon.SinonSandbox;
+    let clientStub: sinon.SinonStubbedInstance<RSPClient> = new RSPClient('somehost', 8080);;
+    let serverExplorer: ServersViewTreeDataProvider;
+
+    const serverType: Protocol.ServerType = {
+        description: 'a type',
+        id: 'type',
+        visibleName: 'the type'
+    };
+
+    const serverHandle: Protocol.ServerHandle = {
+        id: 'id',
+        type: serverType
+    };
+
+    const stateChange: Protocol.ServerStateChange = {
+        server: serverHandle,
+        state: 0
+    };
+
+    const ProcessOutput: Protocol.ServerProcessOutput = {
+        processId: "process id",
+        server: serverHandle,
+        streamType: 0,
+        text: "the type"
+    };
+
+    const findServerBeans = {
+        length: 1,
+        fullVersion: "version",
+        location: "path",
+        name: "EAP",
+        serverAdapterTypeId: "org.jboss",
+        specificType: "EAP",
+        typeCategory: "EAP",
+        version: "7.1"
+    }
+
+    const status = {
+        code: 0,
+        message: "ok",
+        pluginId: "unknown",
+        severity: 0
+    }
+
+    test('should able to insert the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        sandbox = sinon.createSandbox();
+        const refreshStub = sandbox.stub(serverExplorer, 'refresh').returns(null);
+        serverExplorer.insertServer(serverHandle);
+        expect(refreshStub).calledOnce;
+    });
+
+    test('should able to removeServer the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const disposeStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+            dispose: () => {}
+        });
+        sandbox.stub(serverExplorer, 'refresh')
+        serverExplorer.removeServer(serverHandle);
+        expect(disposeStub).calledOnce;
+    });
+
+    test('should able to updateServer the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const serviceStub = sinon.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+        });
+        sandbox.stub(serverExplorer, 'refresh')
+        serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to showOutput of the server ', () => {
```

Comment:
rename it to: 'serverExplorer.showOutput call should show server's output channel'